configmap/parse.go

```diff
@@ -247,11 +247,15 @@ func Parse(data map[string]string, parsers ...ParseFunc) error {
 
 // AsOptionalMap parses the data into the target as a map[string]string, if it exists.
 // The map is represented as a list of key-value pairs with a common prefix.
-func AsOptionalMap(prefix string, target map[string]string) ParseFunc {
+func AsOptionalMap(prefix string, target **map[string]string) ParseFunc {
```

Comment:
So the reason we have `AsOptional*` helpers is for when the null value have some meaning.  ie. bool - null is the default (set by the operator), vs. an explicit `true` or `false`

But I don't think we need this for a map - since nil/empty maps behave the same when doing safe guarded key lookups (ie. `v, ok := m[k]`)

Thus I think we can drop the double pointer.

My other suggestion is making the function name more explicit that we're doing this special prefix parsing. `AsMap` implies we are taking a string literal and turn it into a go map. (similar to the other `As[Type]` functions)

Maybe `CollectMapEntriesWithPrefix`?

```suggestion
func CollectMapEntriesWithPrefix(prefix string, target *map[string]string) ParseFunc {
```

Reply:
or `CollectEntriesStartingWith`?

Reply:
Done